brewtils/models.py

```diff
@@ -495,11 +495,15 @@ def __init__(self, heartbeat=None, history=None):
         self.history = history or []
 
     def set_status_heartbeat(self, status, max_history=None):
-
-        self.heartbeat = datetime.utcnow()
-        self.history.append(
-            StatusHistory(status=copy.deepcopy(status), heartbeat=self.heartbeat)
-        )
+        if (
+            not self.history
+            or status != "NOT_CONFIGURED"
+            or (status != self.history[-1].status)
+        ):
```

Comment:
The logic should be:
status != "NOT_CONFIGURED"
OR
not self.history
OR
(status == "NOT_CONFIGURED" AND status != self.history[-1].status)